<?xml version="1.0"?>
<launch>

  <!-- Gazebo setup with simulated IMU and GPS -->
  <param name="use_sim_time" value="true" /> 
  <include file="$(find avc_nrc_2020)/launch/traxxas_rc.launch" >
    <arg name="start_x" value="0" />
    <arg name="start_y" value="0" />
    <arg name="start_z" value="0" />
    <arg name="start_yaw" value="0" />
    <arg name="pub_tf" value="true" />
    <arg name="tf_freq" value="100.0" />
    <arg name="blue" value="false" />
  </include>

  <include file="$(find gazebo_ros)/launch/empty_world.launch" />


  <!-- Diagnostic tools -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find avc_nrc_2020)/rviz/localization.rviz"/>


  <!-- robot_localization nodes: -->
  <node pkg="avc_nrc_2020" type="ACK_EKF_step.py"
     name="ekf_localizer_node" output="screen" clear_params="true">
    <param name="gps_offset_angle" value="0"/>
    <param name="frequency" value="10"/>
    <param name="sensor_timeout" value="2.0"/>
    <param name="r" value="0.07"/>
    <param name="l" value="0.36"/>
    <param name="initial_state" value="[0, 0, 0, 0, 0, 0, 0, 0]"/>
  </node>
</launch>
